projects/cds/src/cms-components/merchandising/carousel/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -0,0 +1,51 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { Product } from '@spartacus/core';
+import { CmsComponentData } from '@spartacus/storefront';
+import { Observable, of } from 'rxjs';
+import { distinctUntilChanged, filter, map, switchMap } from 'rxjs/operators';
+import { CmsMerchandisingCarouselComponent } from '../../../../cds-models/cms.model';
+import { StrategyConnector } from './../../../../merchandising/connectors/strategy/strategy.connector';
+
+@Component({
+  selector: 'cx-merchandising-carousel',
+  templateUrl: './merchandising-carousel.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MerchandisingCarouselComponent {
+  private componentData$: Observable<
+    CmsMerchandisingCarouselComponent
+  > = this.componentData.data$.pipe(filter(Boolean));
+
+  title$: Observable<string> = this.componentData$.pipe(
+    map(data => data.title)
+  );
+
+  size$: Observable<string> = this.componentData$.pipe(
+    map(data => data.numberToDisplay)
+  );
+
+  backgroundColor$: Observable<string> = this.componentData$.pipe(
+    map(data => data.backgroundColour)
+  );
+
+  textColor$: Observable<string> = this.componentData$.pipe(
+    map(data => data.textColour)
+  );
+
+  items$: Observable<Observable<Product>[]> = this.componentData$.pipe(
+    map(data => data.strategy),
+    distinctUntilChanged(),
+    switchMap(strategyId =>
+      this.strategyConnector.loadProductsForStrategy(strategyId)
+    ),
+    map(merchandisingProducts => merchandisingProducts.products),
+    map(products => products.map(product => of(product)))
+  );
+
+  constructor(
+    protected componentData: CmsComponentData<
+      CmsMerchandisingCarouselComponent
+    >,
+    protected strategyConnector: StrategyConnector
```

Comment:
instead of injecting the connector directly here, let's create a facade layer to align what we do in Spartacus even though we're not using the store. 

Basically, I think we should inject the connector in that facade layer and delegate the call here

Reply:
Ok, I've created a CdsMerchandisingProductService that is used instead. Is this what you meant?